roxie/roxiemem/roxiemem.hpp

```diff
@@ -232,7 +232,7 @@ class roxiemem_decl DataBufferBottom : public HeapletBase
 private:
     friend class CDataBufferManager;
     CDataBufferManager * volatile owner;
-    atomic_t okToFree;
+    std::atomic_uint okToFree;      // use uint since it is more efficient on some architectures
```

Comment:
Maybe so but it leads to confusing code...

Would it be better to add a 

```
typedef atomic_bool std::atomic_uint;
```

Then use that. You could even change the typedef according to what was more efficient on a given architecture that way.